### PR TITLE
Fix SSL Proxy (develop)

### PR DESCRIPTION
**Pull request by Gee19**
Refs #SUP-45

### CODE REVIEW

**Comment by funkaoshi** [comment]
Should just squash the two commits down. Seems fine.

File: sdetools/extlib/http_req.py [hunk 1]
```diff
@@ -209,7 +209,12 @@ def connect(self):
                                           ca_certs=self.ca_certs)
         if self.cert_reqs & ssl.CERT_REQUIRED:
             cert = self.sock.getpeercert()
-            hostname = self.host.split(':', 0)[0]
+
+            if self._tunnel_host:
```
Comment (shawnburant) [inline sdetools/extlib/http_req.py]: if you move this logic to a method ```self._get_hostname()``` then you can create a unit test on just this functionality, and mock it out when you test the outer method.

**Comment by funkaoshi** [comment]
@shawnburant's suggestion was a good one. We don't do any unit testing right now, the tests here are all big integration tests. Still good to take that approach.

**Comment by funkaoshi** [comment]
Let's get this merged today.

**Comment by shawnburant** [comment]
Much cleaner, 👍

**Pull request merged**